### ISSUE TITLE
OtrEngineImplTest sometimes fails

### ISSUE THREAD

**Issue by tribut** [ibauersachs/otr4j#14]
While writing the patch for fragmentation, maven would sometimes fail to build the package because of a test failure in a region of code I did not touch. Just running the build again seem to fix it (until it happened again):

```
-------------------------------------------------------------------------------
Test set: net.java.otr4j.OtrEngineImplTest
-------------------------------------------------------------------------------
Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.296 sec <<< FAILURE!
testSession2(net.java.otr4j.OtrEngineImplTest)  Time elapsed: 0.043 sec  <<< ERROR!
java.lang.ArrayIndexOutOfBoundsException: 19
        at net.java.otr4j.crypto.OtrCryptoEngineImpl.sign(OtrCryptoEngineImpl.java:295)
        at net.java.otr4j.session.AuthContextImpl$MessageFactoryImpl.getRevealSignatureMessage(AuthContextImpl.java:111)
        at net.java.otr4j.session.AuthContextImpl.handleDHKeyMessage(AuthContextImpl.java:652)
        at net.java.otr4j.session.AuthContextImpl.handleReceivingMessage(AuthContextImpl.java:428)
        at net.java.otr4j.session.SessionImpl.transformReceiving(SessionImpl.java:472)
        at net.java.otr4j.OtrEngineImpl.transformReceiving(OtrEngineImpl.java:83)
        at net.java.otr4j.OtrEngineImplTest.startSessionWithQuery(OtrEngineImplTest.java:215)
        at net.java.otr4j.OtrEngineImplTest.testSession2(OtrEngineImplTest.java:157)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at junit.framework.TestCase.runTest(TestCase.java:168)
        at junit.framework.TestCase.runBare(TestCase.java:134)
        at junit.framework.TestResult$1.protect(TestResult.java:110)
        at junit.framework.TestResult.runProtected(TestResult.java:128)
        at junit.framework.TestResult.run(TestResult.java:113)
        at junit.framework.TestCase.run(TestCase.java:124)
        at junit.framework.TestSuite.runTest(TestSuite.java:243)
        at junit.framework.TestSuite.run(TestSuite.java:238)
        at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:83)
        at org.apache.maven.surefire.junit4.JUnit4TestSet.execute(JUnit4TestSet.java:53)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:123)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:104)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:164)
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:110)
        at org.apache.maven.surefire.booter.SurefireStarter.invokeProvider(SurefireStarter.java:175)
        at org.apache.maven.surefire.booter.SurefireStarter.runSuitesInProcessWhenForked(SurefireStarter.java:107)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:68)
```

**Comment by tribut** [comment]
This might be fixed by https://github.com/redsolution/otr4j/commit/88ffac00f9275549833108dfa0b41d824d8f787b.

**Comment by dirkmoors** [comment]
I'm experiencing the same issue here:

```
Exception in thread "Thread-4" java.lang.ArrayIndexOutOfBoundsException: 19
    at net.java.otr4j.crypto.OtrCryptoEngineImpl.sign(OtrCryptoEngineImpl.java:295)
    at net.java.otr4j.session.AuthContextImpl$MessageFactoryImpl.getSignatureMessage(AuthContextImpl.java:152)
    at net.java.otr4j.session.AuthContextImpl.handleRevealSignatureMessage(AuthContextImpl.java:615)
    at net.java.otr4j.session.AuthContextImpl.handleReceivingMessage(AuthContextImpl.java:431)
    at net.java.otr4j.session.SessionImpl.transformReceiving(SessionImpl.java:472)
    at net.java.otr4j.session.DummyClient$MessageProcessor.process(DummyClient.java:109)
    at net.java.otr4j.session.DummyClient$MessageProcessor.run(DummyClient.java:130)
    at java.lang.Thread.run(Thread.java:724)
```

Got this traceback by simply running the SessionImplTest.

**Comment by dirkmoors** [comment]
Actually, I ran into this issue multiple times, when trying to implement multithreaded clients...

**Comment by gpolitis** [comment]
It should be fixed in the latest otr4j (0.21). Please let me know if the problem persists.
